controllers/providers/assistant/gslb.go

```diff
@@ -60,27 +61,42 @@ func NewGslbAssistant(client client.Client, k8gbNamespace, edgeDNSServer string)
 
 // CoreDNSExposedIPs retrieves list of IP's exposed by CoreDNS
 func (r *GslbLoggerAssistant) CoreDNSExposedIPs() ([]string, error) {
+	serviceList := &corev1.ServiceList{}
 	coreDNSService := &corev1.Service{}
-	err := r.client.Get(context.TODO(),
-		types.NamespacedName{Namespace: r.k8gbNamespace, Name: coreDNSExtServiceName}, coreDNSService)
+	sel, err := labels.Parse(coreDNSServiceLabel)
+	if err != nil {
+		log.Err(err)
+	}
+	listOption := &client.ListOptions{
+		LabelSelector: sel,
+		Namespace:     r.k8gbNamespace,
+	}
+
+	err = r.client.List(context.TODO(), serviceList, listOption)
 	if err != nil {
 		if errors.IsNotFound(err) {
-			log.Warn().Msgf("Can't find %s service", coreDNSExtServiceName)
+			log.Warn().Msgf("Can't find service %s", err.Error())
 		}
+	}
+	if len(serviceList.Items) != 1 {
+		log.Info().Msgf("items are not 1 actual %d", len(serviceList.Items))
+		err := coreerrors.New("Can't find CoreDNS service")
```

Comment:
Error string should not be capitalized, see: https://github.com/golang/go/wiki/CodeReviewComments#error-strings